indico/htdocs/js/indico/jquery/rulelistwidget.js

```diff
@@ -0,0 +1,213 @@
+            ruleContext.$element.find('.rule-condition').show();
+
+            this.conditionSpec.forEach(function(condition) {
+                var $condition = ruleContext.$element.find('.rule-condition[data-condition=' + condition.value + ']');
+                var newOption = $condition.find('select').val();
+
+                if (condition.compatibleWith) {
+                    var compatibleConditions = condition.compatibleWith[newOption];
+                    ruleContext.$element.find('.rule-condition').each(function() {
+                        var $this = $(this);
+                        var conditionName = $this.data('condition');
+                        if (this === $condition.get(0)) {
+                            return;
+                        }
+                        if (_.contains(compatibleConditions, conditionName)) {
+                            $this.show();
+                        } else {
+                            $this.hide();
+                            delete ruleContext.rule[conditionName];
+                        }
+                    });
+                }
+            });
+        },
+
+        _setConditionOption: function(ruleContext, conditionName, value) {
+            var $condition = ruleContext.$element.find('.rule-condition[data-condition=' + conditionName + ']');
+            var $select = $condition.find('select');
+
+            if (value === undefined) {
+                value = this.options.conditionChoices[conditionName].options[0][0];
+            }
+
+            // will implicitly call _optionSelected(...)
+            $select.val(_optionValue(value));
+        },
+
+        _checkPlaceholder: function() {
+            var $placeholder = this.options.containerElement.find('.no-rules-placeholder');
+            if (this.ruleData.length) {
+                $placeholder.hide();
+            } else {
+                $placeholder.show();
+            }
+        },
+
+        _addRule: function($ruleList, rule) {
+            var self = this;
+            var ruleContext = {
+                $element: $('<li>', {class: 'rule'}),
+                rule: rule
+            };
+            var $removeButton = $('<a>', {class: 'i-button-icon icon-remove rule-remove-button'});
+
+            ruleContext.$element.append(
+                _.map(this.conditionSpec, function(condition) {
+                    var $html = self._renderRuleCondition(ruleContext, condition);
+                    return $html;
+                }));
+            ruleContext.$element.append($removeButton).appendTo($ruleList);
+            ruleContext.$element.data('ruleContext', ruleContext);
+
+            this.conditionSpec.forEach(function(condition) {
+                self._setConditionOption(ruleContext, condition.value, rule[condition.value]);
+            });
+            this._checkConditionIncompatibility(ruleContext);
+            this._checkPlaceholder();
+            this.element.trigger('change');
```

Comment:
wouldn't it be better to do this from the place where we also call `this.element.val(..)`